src/include/units/physical/si/base/thermodynamic_temperature.h

```diff
@@ -95,8 +101,57 @@ namespace unit_constants {
 inline constexpr auto K = thermodynamic_temperature<kelvin, one_rep>{};
 inline constexpr auto mK = thermodynamic_temperature<millikelvin, one_rep>{};
 
+inline constexpr auto zp_kelvin = thermodynamic_temperature_point<kelvin, zero_rep, kelvin_temperature_origin>{};
+
 inline constexpr auto deg_C = thermodynamic_temperature<degree_celsius, one_rep>{};
 
+inline constexpr auto zp_deg_celsius = thermodynamic_temperature_point<degree_celsius, zero_rep, celsius_temperature_origin>{};
```

Comment:
I am not sure if it works or is enough (see my proposal for an example application). Analyzing code in code reviews without the possibility to play with it is not the easiest thing :-( This is why I want to ensure that the below is possible with the current design.

I believe that for the Celsius quantity point it does not matter how it relates to Kelvin as long as we do not have to make arithmetic or comparison on those two units.

Celsius is defined in terms of Kelvin so, in my opinion, the definition of a quantity_point should also provide the offset to Kelvin scale default origin. This should be always used when converting between different units.

Another offset is the offset provided for the user to define its quantity_point (i.e. 21oC in my example). It should be any value that the user wants and should not override the offset needed to convert units.

The above two offsets should be independent.

Reply:
You will have to clarify further, I do not really understand what you are saying here. I can say that it does work according to (my) expectations.

In this design, a quantity_point can be considered to be represented as a sum of an "offset" (with the semantics of a quantity), which is supplied by the user as the constructor argument, and an abstract "origin" (with the semantics of a quantity_point again), represented by the `Orig` template argument. In the following, I will use lower-case letters to represent symbolic objects with quantity semantics and upper-case letters to represent symbolic objects with quantity_point semantics.

Assume you have two abstract origins, call them `A` and `B` on the same dimension scale, and you happen to know the distance between them: `d = A-B` (differences between quantity_points are quantities). You can express that distance in any unit of the underlying dimension. You can now also re-express the abstract origin quantity_point `A` as  `A = d+B`, that is, referred to a reference origin `B`. If the user now has a quantity_point `X` with offset `a` with respect to `A`, internally represented as `a+A`, you can just substitute that expression and find that `X = a + d + B = (a+d) + B`. Therefore, with `b = a+d`, you have found now the new expression for the same quantity_point with respect to a different origin as  `X = b + B`.

This is what this implementation does, where the model of the `PointOrigin` `A` may optionally store the symbolic identity of `B` (in the type alias `reference_origin`) and the distance of `d` (in the static variable `offset_to_reference`). After this write-down I realise that it could equally well be represented using a single static variable of type quantity_point, representing `(d+B)`.

Reply:
What I am trying to say is that `assert(quantity_point(1 * deg_C) == quantity_point(1 * K))` should always pass. I am not sure why it collides with CTAD or a single-argument constructor?

`d` between origins of `K` and `C` is well defined and fixed. It should matter only if more than one unit is being used in the same expression (either in comparison or arithmetic) forcing us to convert units. Otherwise, such an offset `d` is irrelevant and can only slow down runtime computations.

Additionally, Celsius and Fahrenheit are defined in terms of Kelvin so also Kelvin should be used to define their origins. 

Initially, when I was planning at some point to add this feature I thought it will be a small extension to `quantity_point_cast`. I thought there will be a variable/type trait that a user will have to specialize for a unit if its origin is different than the origin of the reference unit. With this, the definition of the unit should be as simple as:

```cpp
struct degree_fahrenheit : named_scaled_unit<degree_fahrenheit, basic_symbol_text{"\u00b0F", "`F"}, no_prefix, ratio(5, 9), kelvin> {};

template<>
inline constexpr auto origin_offset<degree_fahrenheit> = 459.67 * K;
```

or a bit more advanced with information of when to apply the offset (before unit scaling or after) or another type used instead of quantity. Having this information, `quantity_point_cast` implementation would account for it when casting between different units. Otherwise, when working with the same unit, we do not need this offset at all.

What we have right now in the PR is what I see a lot of additional complexity that I initially did not expect. This is why I want to understand the rationale and ensure that the initial the most important requirements are met.